File: April/Abiji-2020/Golang /Valid Parenthesis String/Valid Parenthesis String.go
func checkValidString(s string) bool {
    left:=0
    right:=0
    for _,c:= range s{
        if c=='('{
            left++;
        }else{
            left--;
        }
        if c==')'{
            right--;
        }else{
            right++;
        }
        if (right<0) {
            break;
        }
        left =int( math.Max(float64(left),float64(0)))
    }
    return left==0
}

File: April/Abiji-2020/C++/Valid Parenthesis String/Valid Parenthesis string.cpp
class Solution {
public:
    bool checkValidString(string s) {
        int left =0, right =0;
        for(char c: s){
            left += c=='('? 1:-1;
            right += c==')'? -1:1;
            if (right<0){
                break;
            }
            left = max(left,0);
        }
        return left ==0;
        
    }
};
